### ISSUE TITLE
Github action fails with `Check could not be completed`

### ISSUE THREAD

**Issue by korya** [linear-b/gitstream#317]
First of all, I would like to use the opportunity and thank you guys for the tool! It is really great! We are evaluating it right now and soon hope to move on and use it in our CI pipeline.

**Describe the bug**

The problem we are experiencing right now is the gitstream check failing with `Check could not be completed` error.

<img width="1110" alt="Screenshot 2023-09-14 at 06 58 34" src="https://github.com/linear-b/gitstream/assets/148461/2e2f8ef7-471b-4e9e-9e53-63342b970b38">

It seems that the job just times out after running for 10 minutes.

The main problem is that the error is not actionable on our side. There is nothing I can do about the error. There is no visibility for why it failed.

The second problem is the long running time -- 10 minutes seems to be ridiculously long to run the basic automation we've setup for our repo.

**To Reproduce**

1. I've just setup gitstream for a repo, configured 5 automations taken from the official docs (5 different `.cm` files):
   1. https://docs.gitstream.cm/automations/approve-safe-changes/
   2. https://docs.gitstream.cm/automations/provide-estimated-time-to-review/
   3. https://docs.gitstream.cm/automations/standard/explain-code-experts/
   4. https://docs.gitstream.cm/automations/integrations/jira/label-missing-jira-info/
   5. https://docs.gitstream.cm/automations/standard/review-assignment/share-knowledge/
2. The PR is for a private repo

**Expected behavior**

First of all, I would expect the job to succeed. But even if it does not I would like to have more visibility as for why it failed. 

**Screenshots**

<img width="1110" alt="Screenshot 2023-09-14 at 06 58 34" src="https://github.com/linear-b/gitstream/assets/148461/2e2f8ef7-471b-4e9e-9e53-63342b970b38">

**Additional context**

It would be great to support some kind of debug/verbose mode when gitstream will share more of its progress/logs in the Github action tab.

**Comment by korya** [comment]
Latest findings are the time out is caused by ill-formatted `includes(regex=...)` call. Below is the sample config for reproducing the error:

```yaml
# vi: set ft=yaml :

manifest:
  version: 1.0

automations:
  label_missing_jira_info:
    if:
      - {{ not (has.test) }}
    run:
      - action: add-label@v1
        args:
          label: 'cm: missing-jira'

has:
  test: {{ pr.description | includes(regex=r/github\.com\/org\/[^/]+\/pull\/\d{1,7}) }}
```

The config problem is in the regexp: `r/github\.com\/org\/[^/]+\/pull\/\d{1,7}`. The slash in `[^/]` part is not escaped.

That being said, there is an error on the gitstream side as well. It should be able to report this kind of errors early instead of timing out.

**Comment by korya** [comment]
Another issue worth mentioning here.

While diagnosing the issue, I commented out template directives `{{ ... }}` with the YAML comment symbol ,`#`, in order to disable running them. I was a bit surprised to discover that the comment did not work and the commented out directive was still executed (and caused the CI job to hang).

Now when I think about it, it seems to be very reasonable that the execution of the template directives ignores YAML files. But while editing the config file, it did not seem that obvious. The IDE treating it as a comment does not help with that either.

The bottom line is consider to avoid interpreting template directives inside YAML comments unless there is a real use case for it.

**Comment by PavelLinearB** [comment]
@korya, thanks for raising the issues.
Could you please share with us the execution logs of gitStream's action?
I would also like to take care of the bug you reported. Could you please share it with me as well?

I am available by email at pavel.vaks@linearb.io

**Comment by korya** [comment]
I would love to share these logs but I am not really sure how to get them. The tool output is not visible in Github actions. Do I need to enable some additional flags?
